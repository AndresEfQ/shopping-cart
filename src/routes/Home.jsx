import styled from "styled-components";
import { Link } from "react-router-dom";

export default function Home() {
  return (
    <Container>
      <Cell>
        <div>
          <p>The place to find all your MTG cards requirements</p>
          <Link to={"/shop"}>
            <button>Shop now</button>
          </Link>
        </div>
      </Cell>
    </Container>
  )
}

const Container = styled.div`
  height: 80%;
  margin-left: 6rem;
  font-size: 4rem;
  color: rgb(255, 255, 255);
  text-shadow: 0 0 10px rgb(0, 0, 0);
  display: flex;
  flex-direction: column;
  justify-content: center;

  & p {
    text-align: center;
  }

  button {
    margin-top: 6vh;
    font-size: 2rem;
    padding: 1rem 6rem;
    border-radius: 20px;
    border: none;
    outline: none;
    cursor: pointer;
    font-weight: 600;
  }
`
const Cell = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);

  & > div {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`;
