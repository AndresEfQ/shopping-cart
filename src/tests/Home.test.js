import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from "../routes/Home";

describe("Home component", () => {

  test("Renders correctly", () => {
    const { container } = render(<Home />, {wrapper: BrowserRouter})
    expect(container).toMatchSnapshot();
  });

  test("Shop call to action has shop link", async () => {
    render(<Home />, {wrapper: BrowserRouter});

    expect(screen.getByRole("link", {name: "Shop now"}).href).toMatch("/shop");
  });

});