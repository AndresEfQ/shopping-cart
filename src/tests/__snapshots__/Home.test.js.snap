// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders correctly 1`] = `
<div>
  <div
    class="sc-beqWaB kpwjKu"
  >
    <div
      class="sc-gueYoa ttHSZ"
    >
      <div>
        <p>
          The place to find all your MTG cards
        </p>
        <a
          href="/shop"
        >
          <button>
            Shop now
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
